Console/Command/NodesValidatorCommand.php

```diff
@@ -0,0 +1,203 @@
+ */
+class NodesValidatorCommand extends Command
+{
+    private MenuRepositoryInterface $menuRepository;
+    private NodeRepositoryInterface $nodeRepository;
+    private SearchCriteriaBuilder $searchCriteriaBuilder;
+    private Validator $validator;
+    private ValidationAggregateError $validationAggregateError;
+    private State $state;
+
+    public function __construct(
+        MenuRepositoryInterface $menuRepository,
+        NodeRepositoryInterface $nodeRepository,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        Validator $validator,
+        ValidationAggregateError $validationAggregateError,
+        State $state,
+        string $name = null
+    ) {
+        $this->menuRepository = $menuRepository;
+        $this->nodeRepository = $nodeRepository;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->validator = $validator;
+        $this->validationAggregateError = $validationAggregateError;
+        $this->state = $state;
+
+        parent::__construct($name);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $description = 'It will loop all menu nodes and return the IDs and description of invalid menu nodes.'
+            . PHP_EOL
+            . ' Additionally, it will ask to remove the detected invalid node if needed.';
+
+        $this->setName('snowdog-menu:validate-nodes')
+            ->setDescription($description);
+
+        parent::configure();
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->setAreaCode();
+        $invalidNodeIds = [];
+
+        foreach ($this->getAllMenus() as $menu) {
+            $output->writeln(PHP_EOL . '<info>... Validating nodes for menu with ID: ' . $menu->getMenuId(). ' </info>');
+            $nodes = $this->nodeRepository->getByMenu($menu->getMenuId());
+            $menuInvalidNodes = $this->getInvalidNodes($nodes);
+
+            if (empty($menuInvalidNodes)) {
+                $output->writeln('<info>No invalid nodes encountered</info>');
+                continue;
+            }
+
+            $output->writeln(
+                '<error>Invalid nodes encountered for menu ID: ' . $menu->getMenuId() . '</error>'
+            );
+
+            foreach ($menuInvalidNodes as $errorMessages) {
+                foreach ($errorMessages as $errorMessage) {
+                    $output->writeln("<error>- $errorMessage</error>");
+                }
+            }
+
+            $invalidNodeIds = array_merge($invalidNodeIds, array_keys($menuInvalidNodes));
+        }
+
+        if (empty($invalidNodeIds) || !$input->isInteractive()) {
+            return;
+        }
+
+        $helper = $this->getHelper('question');
+
+        if (!$helper->ask($input, $output, $this->getConfirmationQuestion())) {
+            $output->writeln(PHP_EOL . '<info>Invalid nodes were NOT removed</info>');
+            return;
```

Comment:
Please check this exception message
`Return value of "Snowdog\Menu\Console\Command\NodesValidatorCommand\Interceptor::execute()" must be of the type int, "null" returned.`